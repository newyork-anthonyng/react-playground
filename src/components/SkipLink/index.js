import React, { Component } from 'react';
import T from 'prop-types';
import glamorous from 'glamorous';

const HiddenLink = glamorous.a({
  backgroundColor: 'black',
  color: 'white',
  padding: '25px',
  position: 'absolute',
  textDecoration: 'none',
  top: '-9999px',
  ':focus': {
    top: 0,
  },
});

class SkipLink extends Component {
  render() {
    const { href, ...rest } = this.props;

    return (
      <HiddenLink
        href={href}
        {...rest}
      >
        Skip to Navigation
      </HiddenLink>
    );
  }
}

SkipLink.propTypes = {
  href: T.string.isRequired,
  '*': T.any,
};

export default SkipLink;
