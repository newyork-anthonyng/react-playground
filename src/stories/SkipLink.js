import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs';
import glamorous from 'glamorous';

const ContainerPage = glamorous.div({
  height: '200vh',
  backgroundColor: 'aliceblue',
  position: 'relative',
});
const Footer = glamorous.div({
  position: 'absolute',
  bottom: 0,
  left: 0,
});

import SkipLink from '../components/SkipLink';

storiesOf('SkipLink', module)
  .addDecorator(withKnobs)
  .add('With knobs', () => {
    return (
      <ContainerPage>
        <SkipLink
          href="#footer"
          onClick={() => console.log('Hello World')}
        />
        <Footer id="footer">
          I am a footer
        </Footer>
      </ContainerPage>
    );
  });
