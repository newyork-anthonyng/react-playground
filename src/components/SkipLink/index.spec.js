import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import SkipLink from './';

const defaultProps = {
  href: '#footer',
};

it('should render correctly', () => {
  const wrapper = mount(
    <SkipLink {...defaultProps} />
  );
  expect(toJSON(wrapper)).toMatchSnapshot();
});
