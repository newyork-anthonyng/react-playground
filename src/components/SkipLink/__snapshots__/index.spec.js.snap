// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<SkipLink
  href="#footer"
>
  <glamorous(a)
    href="#footer"
  >
    <a
      className="css-s1tdzo"
      href="#footer"
    >
      Skip to Navigation
    </a>
  </glamorous(a)>
</SkipLink>
`;
